test/e2e/git_test.go

```diff
@@ -90,3 +98,79 @@ func TestGitSemverResolutionUsingConstraintWithLeadingZero(t *testing.T) {
 		Expect(SyncStatusIs(SyncStatusCodeSynced)).
 		Expect(Pod(func(p corev1.Pod) bool { return strings.HasPrefix(p.Name, "new-app") }))
 }
+
+func TestAnnotatedTagInStatusSyncRevision(t *testing.T) {
```

Comment:
- Your E2E tests are not currently correctly testing the failing case, AFAICT. 
- Notice that if you comment out your fix in this PR (e.g. you remove the changes in `workaround.go`), that those E2E tests still pass.
	- Since the fix has been removed, obviously the tests should fail. If they don't, that means the tests aren't testing what we think they are.
	- The tests in my original E2E test branch do not exhibit this behaviour, if you want to use those as originally constructed.